src/app/billing-client.ts

```diff
@@ -0,0 +1,97 @@
+import { type RpcOptions } from '@improbable-eng/grpc-web/dist/typings/client.d';
+import { BillingServiceClient } from '../gen/app/v1/billing_pb_service';
+import pb from '../gen/app/v1/billing_pb';
+import { promisify } from '../utils';
+
+type GetCurrentMonthUsageResponse =
+  Partial<pb.GetCurrentMonthUsageResponse.AsObject> & {
+    start?: Date;
+    end?: Date;
+  };
+
+const decodeMonthUsage = (
+  proto: pb.GetCurrentMonthUsageResponse
+): GetCurrentMonthUsageResponse => {
+  const startDate = proto.getStartDate();
+  const endDate = proto.getEndDate();
+  const result: GetCurrentMonthUsageResponse = {
+    cloudStorageUsageCost: proto.getCloudStorageUsageCost(),
+    dataUploadUsageCost: proto.getDataUploadUsageCost(),
+    dataEgresUsageCost: proto.getDataEgresUsageCost(),
+    remoteControlUsageCost: proto.getRemoteControlUsageCost(),
+    standardComputeUsageCost: proto.getStandardComputeUsageCost(),
+    discountAmount: proto.getDiscountAmount(),
+    totalUsageWithDiscount: proto.getTotalUsageWithDiscount(),
+    totalUsageWithoutDiscount: proto.getTotalUsageWithoutDiscount(),
+    start: startDate
+      ? new Date(
+          startDate.getSeconds() * 1000 + startDate.getNanos() / 1_000_000
+        )
+      : undefined,
+    end: endDate
+      ? new Date(endDate.getSeconds() * 1000 + endDate.getNanos() / 1_000_000)
+      : undefined,
+  };
+  return result;
+};
```

Comment:
Do you really need all of this? I think `proto.getStartDate()` has a `toDate()` function that you can use instead.

Reply:
Thank you for reminding me! I was using it before, but it was failing for something else and so I did it manually. I forgot to change it back.